components/template-mgt/org.wso2.carbon.identity.template.mgt/src/main/java/org/wso2/carbon/identity/template/mgt/dao/impl/TemplateManagerDAOImpl.java

```diff
@@ -0,0 +1,238 @@
+import org.wso2.carbon.identity.template.mgt.util.JdbcUtils;
+import org.wso2.carbon.identity.template.mgt.util.TemplateMgtUtils;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+
+import static org.wso2.carbon.identity.template.mgt.TemplateMgtConstants.ErrorMessages.ERROR_CODE_DELETE_TEMPLATE;
+import static org.wso2.carbon.identity.template.mgt.TemplateMgtConstants.ErrorMessages.ERROR_CODE_INSERT_TEMPLATE;
+import static org.wso2.carbon.identity.template.mgt.TemplateMgtConstants.ErrorMessages.ERROR_CODE_LIST_TEMPLATES;
+import static org.wso2.carbon.identity.template.mgt.TemplateMgtConstants.ErrorMessages.ERROR_CODE_SELECT_TEMPLATE_BY_NAME;
+import static org.wso2.carbon.identity.template.mgt.TemplateMgtConstants.ErrorMessages.ERROR_CODE_UPDATE_TEMPLATE;
+import static org.wso2.carbon.identity.template.mgt.TemplateMgtConstants.SqlQueries.DELETE_TEMPLATE;
+import static org.wso2.carbon.identity.template.mgt.TemplateMgtConstants.SqlQueries.GET_TEMPLATE_BY_NAME;
+import static org.wso2.carbon.identity.template.mgt.TemplateMgtConstants.SqlQueries.LIST_PAGINATED_TEMPLATES_DB2;
+import static org.wso2.carbon.identity.template.mgt.TemplateMgtConstants.SqlQueries.LIST_PAGINATED_TEMPLATES_MSSQL;
+import static org.wso2.carbon.identity.template.mgt.TemplateMgtConstants.SqlQueries.LIST_PAGINATED_TEMPLATES_MYSQL;
+import static org.wso2.carbon.identity.template.mgt.TemplateMgtConstants.SqlQueries.LIST_PAGINATED_TEMPLATES_ORACLE;
+import static org.wso2.carbon.identity.template.mgt.util.JdbcUtils.isDB2DB;
+import static org.wso2.carbon.identity.template.mgt.util.JdbcUtils.isH2MySqlOrPostgresDB;
+import static org.wso2.carbon.identity.template.mgt.util.JdbcUtils.isMSSqlDB;
+
+/**
+ * Perform CRUD operations for {@link Template}.
+ *
+ * @since 1.0.0
+ */
+public class TemplateManagerDAOImpl implements TemplateManagerDAO {
+
+    /**
+     * Add a {@link Template}.
+     *
+     * @param template {@link Template} to insert.
+     * @return Inserted {@link TemplateInfo}.
+     * @throws TemplateManagementException If error occurs while adding the {@link Template}.
+     */
+    public TemplateInfo addTemplate(Template template) throws TemplateManagementException {
+
+        TemplateInfo templateResult;
+        JdbcTemplate jdbcTemplate = JdbcUtils.getNewTemplate();
+
+        try {
+            jdbcTemplate.executeUpdate(TemplateMgtConstants.SqlQueries.INSERT_TEMPLATE, (preparedStatement -> {
+                preparedStatement.setInt(1, template.getTenantId());
+                preparedStatement.setString(2, template.getTemplateName());
+                preparedStatement.setString(3, template.getDescription());
+                try {
+                    InputStream inputStream = IOUtils.toInputStream(template.getTemplateScript());
+                    preparedStatement.setBinaryStream(4, inputStream, inputStream.available());
+                } catch (IOException e) {
+                    e.printStackTrace();
```

Comment:
don't use e.printStackTrace();. Use a logger and log the error with a proper error message.